sdk/keyvault/Azure.Security.KeyVault.Administration/src/RoleAssignmentScope.cs

```diff
@@ -28,10 +28,20 @@ public RoleAssignmentScope(string value)
         /// <summary>
         /// Initializes a new instance of the <see cref="RoleAssignmentScope"/> structure.
         /// </summary>
-        /// <param name="ResourceId">The Resource Id for the given Resource</param>
+        /// <param name="ResourceId">The Resource Id for the given Resource.</param>
         public RoleAssignmentScope(Uri ResourceId)
         {
-            _value = ResourceId.AbsolutePath ?? throw new ArgumentNullException(nameof(ResourceId));
+            _ = ResourceId ?? throw new ArgumentNullException(nameof(ResourceId));
+
+            // Remove the version segment from a Key Id, if present.
+            if (ResourceId.AbsolutePath.StartsWith("/keys/", StringComparison.Ordinal) && ResourceId.Segments.Length == 4)
```

Comment:
Every time you access `Segments`, it allocates a new array and parses `AbsolutePath` into it (effectively). Store it as a local once to avoid some of the allocations here.